swift/apple/FirezoneKit/Sources/FirezoneKit/Managers/TunnelManager.swift

```diff
@@ -236,6 +230,27 @@ public class TunnelManager {
       options = ["token": token as NSObject]
     }
 
+#if os(macOS)
+    // On macOS we use System Extensions, and we need to wait for them to be activated
+    // before we can continue starting the tunnel. Otherwise, the tunnel will come up,
+    // but then be killed immediately after since the system will reap its process as
+    // the system extension is moved in place. This is more of an issue in development
+    // where the system will replace the extension on each call to this API, ignoring the
+    // version check that typically prevents extensions from being reactivated if they have the
+    // same marketing version.
+    systemExtensionManager.installSystemExtension(identifier: bundleIdentifier) { error in
+      if let error = error {
+        Log.app.error("\(#function): Installing system extension failed! \(error.localizedDescription)")
+      } else {
+        self.startTunnel(options: options)
+      }
```

Comment:
```suggestion
      if let error = error {
        Log.app.error("\(#function): Installing system extension failed! \(error.localizedDescription)")
        return
      }
      
      self.startTunnel(options: options)
```

Personally, I find that avoiding `else` makes code more readable because it visually separates happy-path from error paths through differences in indentation.

Reply:
Unfortunately this just returns from the closure and not the outer function.

Reply:
Actually that shouldn't matter in this case!

Reply:
A closure should have a scoped control flow so you should be able to early return from that to avoid executing statements after the `if`.